examples/oneview_switch_facts.yml

```diff
@@ -0,0 +1,45 @@
+###
+# Copyright (2016) Hewlett Packard Enterprise Development LP
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# You may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+###
+---
+- hosts: all
+  vars:
+    - config: "{{ playbook_dir }}/oneview_config.json"
+    - name: "172.18.20.1" # change for the desired switch name
+  tasks:
+    - name: Gather facts about all switches
+      oneview_switch_facts:
+        config: "{{ config }}"
+      delegate_to: localhost
+
+    - debug: var=switches
+
+    - name: Gather facts about the switch that matches the specified switch name
+      oneview_switch_facts:
+        config: "{{ config }}"
+        name: "{{ name }}"
+      delegate_to: localhost
+
+    - debug: var=switches
+
+    - name: Gather facts about the environmental configuration for the switch that matches the specified switch name
+      oneview_switch_facts:
+        config: "{{ config }}"
+        name: "{{ name }}"
+        options:
+          - environmental_configuration
```

Comment:
Yes! In my opinion, we should use snake case for most of cases. Camel case should be used only for properties that will be sent to OneView.